### PR TITLE
Implement a locking variable to circumvent a infinite regression

### LINKED ISSUE #2876: Infinite loop when using Entity->close() during EntityDespawnEvent
Plugin developer stupidity could lead to this circumstance which can be problematic, because `Entity->closed` is not set to true until after the event.

### PR DESCRIPTION
**Pull request by geNAZt**
## Introduction
The current problem is that a plugin could enter a infinite loop when calling Entity::close in a EntityDespawnEvent.

### Relevant issues

* Fixes #2876

### API changes
Entity::close may not fire any additional events when called from inside the EntityDespawnEvent. 

## Backwards compatibility
I haven't touched the $closed ordering to fix this issue since it would break the event contract on the entity. That should maybe be done on 4.0 but not on 3.x

## Follow-up
Think about reordering close variable state inside Entity::close

### CODE REVIEW

File: src/pocketmine/entity/Entity.php [hunk 1]
```diff
@@ -2108,7 +2110,12 @@ public function isClosed() : bool{
 	 * WARNING: Entities are unusable after this has been executed!
 	 */
 	public function close() : void{
+		if($this->closeInFlight) {
```
Comment (dktapps) [inline src/pocketmine/entity/Entity.php]: just one nit here (formatting)

Comment (geNAZt) [reply src/pocketmine/entity/Entity.php]: Tell me pls :(

Comment (dktapps) [reply src/pocketmine/entity/Entity.php]: pls remove the space between ) and {

**Comment by dktapps** [comment]
you just introduced a bunch of unrelated whitespace changes.

**Comment by lukeeey** [comment]
like all his prs lol

**Pull request merged**